### PR TITLE
adds color picker for JSON editor

### PR DESCRIPTION
**Pull request by RaphaelAlvez**
Created code based on other commands

context based on image upload
options based on duplicate
setting property based on toggle boolean

No example available for populating input (color always start at #ff0000)

### CODE REVIEW

File: client/js/jsonedit.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ const jeCommands = [
     id: 'je_colorPicker',
     name: 'change color',
     options: [ { type: 'color', label: 'color' } ],
-    context: '.*#([0-9a-fA-F]{8}|[0-9a-fA-F]{6}|[0-9a-fA-F]{4}|[0-9a-fA-F]{3})|^.* ↦ color$' + String.fromCharCode(36),
+    context: '.*#([0-9a-fA-F]{8}|[0-9a-fA-F]{6}|[0-9a-fA-F]{4}|[0-9a-fA-F]{3})|^.* ↦ color',
```
Comment (ArnoldSmith86) [inline client/js/jsonedit.js]: Correct should be the middle ground: `'.*#([0-9a-fA-F]{8}|[0-9a-fA-F]{6}|[0-9a-fA-F]{4}|[0-9a-fA-F]{3})|^.* ↦ color' + String.fromCharCode(36)`

Doesn't really matter though if the context ends with `color` or if something is after that so :shrug:.

Comment (RaphaelAlvez) [reply client/js/jsonedit.js]: > > No example available for populating input (color always start at #ff0000)
> 
> If you want to preset the input with the current value of the property you're editing, then I don't think that's possible yet.

then this PR is ready. it's already better then what we currently have

**Comment by bjalder26** [comment]
I don't fully understand this portion of the code but one of the things I noticed is that once the color picker is open it doesn't automatically close until you click cancel or go it just stays in the command options.  This does not really interfere with its operation, the only real effect is that the color picker could then be used with other properties by clicking out of the color line.

It works, and I don't see any possibility of it breaking games.  I also don't think it is extremely important that it is prepopulated with the original color because it is not part of a larger input, and the only reason to press go is to change the color.

**Comment by rogerl50** [comment]
It's unfortunate that the color wheel always shows up at bottom left. Can it be in the lower right, closer to the "Change color" button?

**Comment by bjalder26** [comment]
> It's unfortunate that the color wheel always shows up at bottom left. Can it be in the lower right, closer to the "Change color" button?

Which browser?

Chrome:
image

Firefox:
image

Edge:
image

**Comment by bjalder26** [comment]
> Firefox on Mac

Oh... I wonder if something like `style="text-align: center"` or `style="margin-left:650px"` would align it.

https://stackoverflow.com/questions/30515873/how-to-set-the-default-location-of-the-popup-color-picker-html-5

**Pull request merged**